src/debot/interfaces/amount_input.rs

```diff
@@ -0,0 +1,95 @@
+use crate::debot::term_browser::terminal_input;
+use serde_json::Value;
+use ton_client::abi::Abi;
+use ton_client::debot::{DebotInterface, InterfaceResult};
+use super::dinterface::{decode_answer_id, decode_num_arg, decode_prompt};
+use ton_client::encoding::decode_abi_number;
+use crate::convert;
+
+const ID: &'static str = "a1d347099e29c1624c8890619daf207bde18e92df5220a54bcc6d858309ece84";
+
+pub const ABI: &str = r#"
+{
+	"ABI version": 2,
+	"header": ["time"],
+	"functions": [
+		{
+			"name": "get",
+			"inputs": [
+				{"name":"answerId","type":"uint32"},
+				{"name":"prompt","type":"bytes"},
+				{"name":"decimals","type":"uint8"},
+				{"name":"min","type":"uint128"},
+				{"name":"max","type":"uint128"}
+			],
+			"outputs": [
+				{"name":"value","type":"uint128"}
+			]
+		}
+	],
+	"data": [
+	],
+	"events": [
+	]
+}
+"#;
+
+pub struct AmountInput {}
+
+impl AmountInput {
+    pub fn new() -> Self {
+        Self {}
+    }
+    fn get(&self, args: &Value) -> InterfaceResult {
+        let answer_id = decode_answer_id(args)?;
+        let prompt = decode_prompt(args)?;
+        let decimals = decode_num_arg::<usize>(args, "decimals")?;
+        if decimals > 255 {
```

Comment:
if decimals if really decimal numbers then 255 is very great amount. 2^256 ~ 10^77

Reply:
Yep, you are right. But according to interface ABI, `decimals` has `uint8` type and this check only validate it range.